src/tools/rbd/action/Ubbd.cc

```diff
@@ -0,0 +1,189 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#include "common/errno.h"
+#include "tools/rbd/Shell.h"
+#include "tools/rbd/Utils.h"
+#include <iostream>
+#include <boost/algorithm/string/predicate.hpp>
+#include <boost/scope_exit.hpp>
+
+#if defined(WITH_RBD_UBBD)
+extern "C" {
+#include <libubbd.h>
+}
+#endif
+
+namespace rbd {
+namespace action {
+namespace ubbd {
+
+namespace at = argument_types;
+namespace po = boost::program_options;
+
+int execute_list(const po::variables_map &vm,
+                 const std::vector<std::string> &ceph_global_init_args) {
+#if defined(WITH_RBD_UBBD)
+  struct ubbdd_mgmt_rsp rsp = { 0 };
+  struct ubbd_list_options list_opts = { .type = UBBD_DEV_TYPE_RBD };
+  int r;
+  int i;
+
+  r = ubbd_list(&list_opts, &rsp);
+  if (r)
+    return r;
+
+  for (i = 0; i < rsp.list.dev_num; i++) {
+    std::cout << "/dev/ubbd" << rsp.list.dev_list[i] << std::endl;
```

Comment:
Yes, of course in plan, but not in this original PR :)
(1) call ubbd_list to get all dev_id.
(2) call ubbd_info to get detail for each device.
(3) show the result in execute_list()

Reply:
> Yes, of course in plan, but not in this original PR :)

Why not?  Is there a blocker?

If someone were to try `rbd device map -t ubbd` as is, how would they differentiate between ubbd devices if more than one is mapped?

Reply:
No there is no blocker, Just because I want to get the first simple version get merged firstly, and then another patch to improve output of list, even more to support --format and --pretty-format.

Reply:
With Ceph's `TextTable` class, `--format` and `--pretty-format` support is provided automatically.  I don't see a reason to defer what is likely a 50-100 LOC change to a future PR.

Reply:
Hmm, sounds right, lets add it in this PR, thanx